src/common/merge.ts

```diff
@@ -7,30 +7,53 @@ import {
 import { parseInputMergeMethod } from '../utilities/inputParsers';
 import { logDebug, logInfo } from '../utilities/log';
 
+export interface PullRequestDetails {
+  commitHeadline: string;
+  pullRequestId: string;
+  reviewEdge: { node: { state: string } } | undefined;
+}
+
+const EXPONENTIAL_BACKOFF = 2;
+const DEFAULT_WAIT_TIME = 1000;
+
+const delay = async (duration: number): Promise<void> => {
+  return new Promise((resolve: () => void): void => {
+    setTimeout((): void => {
+      resolve();
+    }, duration);
+  });
+};
+
 /**
  * Approves and merges a given Pull Request.
  */
 export const merge = async (
   octokit: ReturnType<typeof getOctokit>,
-  {
-    commitHeadline,
-    pullRequestId,
-    reviewEdge,
-  }: {
-    commitHeadline: string;
-    pullRequestId: string;
-    reviewEdge: { node: { state: string } } | undefined;
-  },
+  pullRequestDetails: PullRequestDetails,
 ): Promise<void> => {
   const mergeMethod = parseInputMergeMethod();
 
+  const { commitHeadline, pullRequestId, reviewEdge } = pullRequestDetails;
+
   const mutation =
     reviewEdge === undefined
       ? approveAndMergePullRequestMutation(mergeMethod)
       : mergePullRequestMutation(mergeMethod);
 
+  await octokit.graphql(mutation, { commitHeadline, pullRequestId });
+};
+
+export const mergeWithRetry = async (
+  octokit: ReturnType<typeof getOctokit>,
+  details: {
+    numberOfRetries: number;
+    trial: number;
+  } & PullRequestDetails,
```

Comment:
I would like to see this intersection definition swapped, so first the `PullRequestDetails`, then the extension.

Reply:
Swapped in d1fdb79